### PR TITLE
Port to NuttX

### PR DESCRIPTION
**Pull request by microhobby**
Let's port the awesome 3BC to NuttX

### CODE REVIEW

**Comment by microhobby** [comment]
@RodrigoDornelles I think this is good to review, please let me know about the documentation if you think that is best also add on this pull request.

File: src/3bc_macros.h [hunk 1]
```diff
@@ -27,6 +27,10 @@
 #define _3BC_MOS6502
 #endif
 
+#if defined(__nuttx__)
```
Comment (RodrigoDornelles) [inline src/3bc_macros.h:30]: Is there any other compilation constants besides the one defined by the programmer? (such as from the nuttx kernel itself)

File: src/3bc_optional_interpreter.h [hunk 2]
```diff
@@ -37,17 +37,28 @@ bool interpreter_3bc_tokens(char* line, char** reg, char** mem, char** val, char
  */
 int interpreter_3bc(app_3bc_t app)
 {  
-    char character = fgetc(app->tty_source.io.stream);
+    int character = fgetc(app->tty_source.io.stream);
+
+#ifdef __nuttx__
```
Comment (RodrigoDornelles) [inline src/3bc_optional_interpreter.h:42]: use constant _3BC_NUTTX

File: src/3bc_optional_interpreter.h [hunk 3]
```diff
@@ -37,17 +37,28 @@ bool interpreter_3bc_tokens(char* line, char** reg, char** mem, char** val, char
...
     if (character == EOF && app->cache_l3.buffer.storage == NULL) {
         return EOF;
     }
 
     /** end of line **/
-    if(character == '\n' || character == '\0' || character == EOF) {
+    if(character == '\n' || character == '\r' || character == '\0' || character == EOF) {
         /** mark end of string **/
         {
+#ifdef __nuttx__
```
Comment (RodrigoDornelles) [inline src/3bc_optional_interpreter.h:57]: use constant _3BC_NUTTX

Comment (RodrigoDornelles) [reply src/3bc_optional_interpreter.h:57]: could be that when not REPL, still print source code, this behavior is not expected.

File: src/driver_tty.c [hunk 4]
```diff
@@ -167,6 +167,7 @@ void driver_tty_output_raw(app_3bc_t app, struct tty_3bc_s tty, const char* stri
     /** stream standard c output **/
     if (tty.type == STREAM_TYPE_COMPUTER_STD){
         fputs(string, tty.io.stream);
+        fflush(tty.io.stream);
```
Comment (RodrigoDornelles) [inline src/driver_tty.c:170]: This may cause unexpected behavior, it may not be necessary.

**Review by RodrigoDornelles — CHANGES_REQUESTED** [review]
There are a few small changes that can be made, as soon as I can compile for nuttx, I will help resolve these requests.

**Review by RodrigoDornelles — APPROVED** [review]

**Pull request merged**